src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -1265,24 +1269,57 @@ public Page search(String q) throws Exception {
     }
 
     /**
-     * Get JSON from A Jenkins endpoint.
-     * @param path The endpoint URL.
-     * @return The JSON.
+     * Get JSON from a Jenkins relative endpoint. Create a new default webclient. If you want to configure the
+     * webclient, for example to set a token for authentication, or accept other HTTP responses than 200, you can use
+     * {@link JenkinsRule#getJSON(String, WebClient)} .
+     * @param path relative path, should not start with '/'
+     * @return The JSON response from server.
      */
     public JSONWebResponse getJSON(@NonNull String path) throws IOException {
-        assert !path.startsWith("/");
 
         JenkinsRule.WebClient webClient = createWebClient();
-        Page runsPage = null;
-        try {
-            runsPage = webClient.goTo(path, "application/json");
-        } catch (SAXException e) {
-            // goTo shouldn't be throwing a SAXException for JSON.
-            throw new IllegalStateException("Unexpected SAXException.", e);
-        }
-        WebResponse webResponse = runsPage.getWebResponse();
+        return getJSON(path, webClient);
+    }
+
+    /**
+     * Get JSON from a Jenkins relative endpoint.
+     * You can preconfigure webClient for example to set a token for authentication, or accept error HTTP status.
+     * @param path relative path, should not start with '/'
+     * @param webClient a pre-configured web client.
+     * @return The JSON response from server.
+     * @throws IOException
+     */
+    public JSONWebResponse getJSON(@NonNull String path, @NonNull JenkinsRule.WebClient webClient) throws IOException {
+        assert !path.startsWith("/");
+
+        URL URLtoCall = new URL(getURL(), path);
+        WebRequest getRequest = new WebRequest(URLtoCall, HttpMethod.GET);
+        getRequest.setAdditionalHeader("Content-Type","application/json");
+        getRequest.setAdditionalHeader("Accept", "application/json");
+        getRequest.setAdditionalHeader("Accept-Encoding", "*");
+
+        return new JSONWebResponse(webClient.loadWebResponse(getRequest));
+    }
+
+    /**
+     * Send JSON content to a Jenkins relative endpoint.
+     * You can preconfigure webClient for example to set a token for authentication, or accept error HTTP status.
+     * @param path relative path, should not start with '/'
+     * @param json the json payload to send
+     * @param webClient a pre-configured web client.
+     * @return The JSON response from server.
+     * @throws IOException
+     */
+    public JSONWebResponse putJSON(@NonNull String path, @NonNull JSON json, @NonNull JenkinsRule.WebClient webClient) throws IOException {
+
+        URL URLtoCall = new URL(getURL(),path);
+        WebRequest putRequest = new WebRequest(URLtoCall, HttpMethod.PUT);
```

Comment:
Seems like a `POST` method would be more useful, as few Jenkins endpoints expect `PUT`?

Reply:
https://github.com/jenkins-infra/jenkins.io/pull/4812#discussion_r787189376

Reply:
There is already a postJSON() method, and I wanted to add a version that allows to pass a webclient also but I couldn't make it work sadly. I can remove PUT from here if it's a blocker, so the get can be merge sooner if you want.

Reply:
update: I have found a way, I hope you will like it :)

Reply:
done.